src/dscanner/analysis/rundmd.d

```diff
@@ -0,0 +1,399 @@
+
+import dscanner.analysis.config : Check, StaticAnalysisConfig;
+import dscanner.analysis.base : BaseAnalyzerDmd, MessageSet;
+
+import dscanner.analysis.alias_syntax_check : AliasSyntaxCheck;
+import dscanner.analysis.always_curly : AlwaysCurlyCheck;
+import dscanner.analysis.asm_style : AsmStyleCheck;
+import dscanner.analysis.assert_without_msg : AssertWithoutMessageCheck;
+import dscanner.analysis.auto_function : AutoFunctionChecker;
+import dscanner.analysis.auto_ref_assignment : AutoRefAssignmentCheck;
+import dscanner.analysis.body_on_disabled_funcs : BodyOnDisabledFuncsCheck;
+import dscanner.analysis.builtin_property_names : BuiltinPropertyNameCheck;
+import dscanner.analysis.constructors : ConstructorCheck;
+import dscanner.analysis.cyclomatic_complexity : CyclomaticComplexityCheck;
+import dscanner.analysis.del : DeleteCheck;
+import dscanner.analysis.enumarrayliteral : EnumArrayVisitor;
+import dscanner.analysis.explicitly_annotated_unittests : ExplicitlyAnnotatedUnittestCheck;
+import dscanner.analysis.final_attribute : FinalAttributeChecker;
+import dscanner.analysis.has_public_example : HasPublicExampleCheck;
+import dscanner.analysis.ifelsesame : IfElseSameCheck;
+import dscanner.analysis.imports_sortedness : ImportSortednessCheck;
+import dscanner.analysis.incorrect_infinite_range : IncorrectInfiniteRangeCheck;
+import dscanner.analysis.label_var_same_name_check : LabelVarNameCheck;
+import dscanner.analysis.lambda_return_check : LambdaReturnCheck;
+import dscanner.analysis.length_subtraction : LengthSubtractionCheck;
+import dscanner.analysis.line_length : LineLengthCheck;
+import dscanner.analysis.local_imports : LocalImportCheck;
+import dscanner.analysis.logic_precedence : LogicPrecedenceCheck;
+import dscanner.analysis.numbers : NumberStyleCheck;
+import dscanner.analysis.objectconst : ObjectConstCheck;
+import dscanner.analysis.opequals_without_tohash : OpEqualsWithoutToHashCheck;
+import dscanner.analysis.pokemon : PokemonExceptionCheck;
+import dscanner.analysis.properly_documented_public_functions : ProperlyDocumentedPublicFunctions;
+import dscanner.analysis.range : BackwardsRangeCheck;
+import dscanner.analysis.redundant_attributes : RedundantAttributesCheck;
+import dscanner.analysis.redundant_parens : RedundantParenCheck;
+import dscanner.analysis.redundant_storage_class : RedundantStorageClassCheck;
+import dscanner.analysis.static_if_else : StaticIfElse;
+import dscanner.analysis.style : StyleChecker;
+import dscanner.analysis.trust_too_much : TrustTooMuchCheck;
+import dscanner.analysis.unmodified : UnmodifiedFinder;
+import dscanner.analysis.unused_label : UnusedLabelCheck;
+import dscanner.analysis.unused_parameter : UnusedParameterCheck;
+import dscanner.analysis.unused_result : UnusedResultChecker;
+import dscanner.analysis.unused_variable : UnusedVariableCheck;
+import dscanner.analysis.useless_assert : UselessAssertCheck;
+import dscanner.analysis.useless_initializer : UselessInitializerChecker;
```

Comment:
why not simply import dscanner.analysis?

Reply:
It's much more cleaner this way. The idea is to organize in the future the src files in analysis dir in other dirs, it will be easier then

Reply:
I don't see how importing dscanner.analysis is less cleaner even if you do shuffle files inside the analysis directory.

Reply:
It's easier to see which files and which declarations are actually used when refactoring. A full `dscanner.analysis` import would also include other modules unrelated to the checks.

Without specific imports, you either have to read the code to verify used imports on rely on compiler errors.

Reply:
This is the main file that uses all the checks. Most of the files in the analysis folder will be needed and all of them implement a single visitor (so you could even make them general imports, not selective ones), so I don't see the argument for spelling them out (and with selective imports too). This to me seems java-level prolixity, however, this really isn't a debate worth our time - do as you wish.